pkg/plugin/trivy/plugin.go

```diff
@@ -839,31 +900,58 @@ func (p *plugin) getPodSpecForClientServerMode(ctx starboard.PluginContext, conf
 			},
 		}
 
-		if _, ok := credentials[container.Name]; ok && secret != nil {
-			registryUsernameKey := fmt.Sprintf("%s.username", container.Name)
-			registryPasswordKey := fmt.Sprintf("%s.password", container.Name)
+		TTLResult, err := config.GetECRRefreshTTL()
+		if err != nil {
+			TTLResult = "0"
+		}
+
+		if config.UseECRCredentials() && CheckAwsEcrPrivateRegistry(container.Image) != "" {
+
+			if utils.TokenTTLValidation(EcrTokenGen, TTLResult) {
```

Comment:
I think its better not to have `EcrTokenGen` and `aws_creds` as global variables, I would, I would create `ECRRegistry` struct as a property of `plugin` which will encapsulate both properties (`EcrTokenGen` and `aws_creds`)

Reply:
Agreed and fixed.